src/engine/QueryPlanner.cpp

```diff
@@ -36,7 +36,8 @@ QueryExecutionTree QueryPlanner::createExecutionTree(ParsedQuery& pq) {
   // from the list of where clause triples. Otherwise the ql:has-relation triple
   // will be handled using a HasRelationScan.
   SparqlTriple patternTrickTriple("", PropertyPath(), "");
-  bool usePatternTrick = checkUsePatternTrick(&pq, &patternTrickTriple);
+  bool usePatternTrick =
+      _enablePatternTrick && checkUsePatternTrick(&pq, &patternTrickTriple);
```

Comment:
I think the comment above makes it sound like this disables the optimized `ql:has-predicate` as well but that's not what we are doing right? I mean then it would be the same as `--no-patterns` but we still want to be able to use `ql:has-predicate` and then compare with the standard SPARQL on the same instance.

Reply:
The HasRelationScan uses the pattern trick data, so `_enablePatternTrick` being false just leads to that being used instead. With `--no-patterns` `ql:has-predicate` stops working entirely and starts throwing exceptions.

Reply:
Ok so it's as I assumed